### ISSUE TITLE
bookdown::word_document2(): Section numbering and header have different styles 

### ISSUE THREAD

**Issue by phargarten2** [rstudio/bookdown#1281]
Thank you for making attractive word documents using **bookdown**. 

As you know, a powerful tool is to create a reference template using **pandoc** and then adjust it [as listed here](https://bookdown.org/yihui/rmarkdown-cookbook/word-template.html) [](https://rmarkdown.rstudio.com/articles_docx.html) . I realize that the different styles in Microsoft Word are utilized by **pandoc** to create finalized report. When I apply `numbering_sections=TRUE`, the numbered sections are in style "Section Number " and the headers are in the sections expected (Header 1, Header 2, Header 3):
![image](https://user-images.githubusercontent.com/17706062/139954766-adfc82f1-b105-40c9-b71e-341b2fc0bc00.png)

I would like the numbered sections to be in the same style as the header. For instance, for header 1, the style to match header 1.  This is what is expected:

![image](https://user-images.githubusercontent.com/17706062/139954813-14bc9812-4247-4e90-9d3f-09cc7d494c21.png)

In the following zip file, the word template, a minimally created R markdown document, and R code to render it  (also pasted below). I also included a word document that consists of what I would expect **pandoc** to create. Thanks!

[bookdown_numbered_section.zip](https://github.com/rstudio/bookdown/files/7464161/bookdown_numbered_section.zip)

```{r}
my_word <- function(...){
  bookdown::word_document2(
    number_sections = TRUE,
    global_numbering = TRUE,
    reference_docx = "word-style-template.docx",
    keep_md = TRUE
  )
}

rmarkdown::render(
  input = "work_formatting.Rmd",
  output_format = my_word(),
  clean = FALSE,
  quiet = FALSE
)
```

```{r}
xfun::session_info('bookdown')
## R version 4.1.1 (2021-08-10)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 19042), RStudio 2021.9.0.351
## 
## Locale:
##   LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
##   LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
##   LC_TIME=English_United States.1252    
## 
## Package version:
##   base64enc_0.1.3  bookdown_0.24.3  bslib_0.3.1      digest_0.6.28    evaluate_0.14   
##   fastmap_1.1.0    fs_1.5.0         glue_1.4.2       graphics_4.1.1   grDevices_4.1.1 
##   highr_0.9        htmltools_0.5.2  jquerylib_0.1.4  jsonlite_1.7.2   knitr_1.36.7    
##   magrittr_2.0.1   methods_4.1.1    R6_2.5.1         rappdirs_0.3.3   rlang_0.4.12    
##   rmarkdown_2.11.3 sass_0.4.0       stats_4.1.1      stringi_1.7.5    stringr_1.4.0   
##   tinytex_0.34.5   tools_4.1.1      utils_4.1.1      xfun_0.27        yaml_2.2.1
```

## Checklist
Welcome to the bookdown GitHub repo !

We are always happy to hear feedback from our users. 

To file a _bug report_, please follow these instructions carefully: <https://yihui.org/issue/#bug-reports>

Also, please complete and keep the checklist below in your issue. This helps you know what to check for opening a good issue report. It also helps know that you've have done the common steps that can solve your potential issue.

At last, if you have posted the same issue elsewhere, please mentioned it (with a link to the other issue).

When filing a _bug report_, please check the boxes below to confirm that you have provided us with the information we need. Have you:

- [X] [formatted your issue](https://yihui.org/issue/#please-format-your-issue-correctly) so it is easier for us to read?

- [X] included a minimal, self-contained, and reproducible example?

- [X] pasted the output from `xfun::session_info('bookdown')` in your issue?

- [X] upgraded all your packages to their latest versions (including your versions of R, the RStudio IDE, and relevant R packages)?

- [X] installed and tested your bug with the development version of the bookdown package using `remotes::install_github("rstudio/bookdown")` ?

**Comment by phargarten2** [comment]
Using the same word template, [I found a different problem regarding the formatting of text in header row of a table](https://github.com/rstudio/rmarkdown/issues/2242). However, I think that the issues are unrelated.

**Comment by cderv** [comment]
Hi  @phargarten2 

Anything related to Word styling using a reference template is done directly by Pandoc, which is used by **rmarkdown**
https://pandoc.org/MANUAL.html#option--reference-doc

We can't really do anything on the **rmarkdown** side regarding styling your document. 

With the current behavior of Pandoc, to get what you want, you would need to modify the Paragraph style Heading 1 and others, to use the character style section number. Or you would need to tweak Section Number character style to be like the character style use withing the paragraph style. If you are not familiar with Word Paragraph and Character Style mechanism, I would suggest looking into this. 

So in a nutshell, you need to tweak the template styles to get what you need as Pandoc will always use Character style "Section Number" for the number before the header. 

Know that Pandoc allow Custom Styles: https://pandoc.org/MANUAL.html#custom-styles and you can also use Lua Filter to tweak its behavior but I am not sure one of them will help in this case. 

Hope it is helpful

We can't do much more on our side about this.

**Comment by phargarten2** [comment]
@cderv, yes, your response is helpful. I am not familiar with word paragraphs and character style mechanisms, and I would need to investigate them.

It would make sense to adjust the section numbering style to match the header. However, depending on whether it is header1, header2, or header3, the font size of the section number would be different. In this case, as I don't think I can have many different section numbering styles, what would you suggest? Is there a way to have this modification within the Microsoft Word Style gallery? Or would you recommend using Pandoc's custom styles?

**Comment by cderv** [comment]
I would advice you to learn about Word styling mechanism and try it by yourself. I think that setting the text style won't impact the font size which is set by the paragraph style.

So I would just try to tweak Word template to your convenience (Currently your template does not tweak Section Number style) and see how it works with Pandoc. What you'll be able to do will be limite to Word features and Pandoc support.

**Comment by phargarten2** [comment]
Thank you @cderv

**Comment by phargarten2** [comment]
@cderv, I wanted to update you and perhaps other users that the modification of Header1, Header2, Header3, and Header4 styles in the reference template did the trick. 
![image](https://user-images.githubusercontent.com/17706062/163213890-1809146e-edf7-4de7-a93e-c8f35d299850.png)

Then, I would turn off `number_sections` to get the desired effect. 

```{r}
 bookdown::word_document2(
    number_sections = FALSE,     #sections are numbered in template to match header style.
    global_numbering = TRUE,
...
)
```

Thanks for your help
